### ISSUE TITLE
Latest indexer-agent Docker image throws error Epoch subgraph could not be queried

### ISSUE THREAD

**Issue by jim-counter** [graphprotocol/indexer#582]
The latest indexer-agent Docker image `ghcr.io/graphprotocol/indexer-agent:v0.20.9` throws the following error when brought `up`:

```
[23:11:04.712] WARN (IndexerAgent/1): Epoch subgraph could not be queried
    component: "NetworkMonitor"
    networkID: "eip155:5"
    networkAlias: "goerli"
    attempt: 1
    retriesLeft: 5
    err: "getaddrinfo EAI_AGAIN api.thegraph.com"
```

Then a number of errors about being unable to query epoch values. Eventually It exhausts its retries and reports:

```
[23:13:50.227] WARN (IndexerAgent/1): Epoch subgraph could not be queried
    component: "NetworkMonitor"
    networkID: "eip155:5"
    networkAlias: "goerli"
    attempt: 6
    retriesLeft: 0
    err: "Failed to query latest valid epoch and block hash"
[23:13:50.227] ERROR (IndexerAgent/1): Failed to query latest epoch number
    component: "NetworkMonitor"
    networkID: "eip155:5"
    networkAlias: "goerli"
    err: {
      "type": "IndexerError",
      "message": "Failed to query latest valid epoch and block hash",
      "stack":
          IndexerError: Failed to query latest valid epoch and block hash
              at indexerError (/opt/indexer/packages/indexer-common/dist/errors.js:169:12)
              at IndexingStatusResolver.<anonymous> (/opt/indexer/packages/indexer-common/dist/indexing-status.js:234:55)
              at Generator.throw (<anonymous>)
              at rejected (/opt/indexer/packages/indexer-common/dist/indexing-status.js:6:65)
              at runMicrotasks (<anonymous>)
              at processTicksAndRejections (node:internal/process/task_queues:96:5)
      "code": "IE070",
      "explanation": "https://github.com/graphprotocol/indexer/blob/main/docs/errors.md#ie070",
      "cause": {
        "type": "Error",
        "message": "Failed to query graph node for blockHashFromNumber: no data returned",
        "stack":
            Error: Failed to query graph node for blockHashFromNumber: no data returned
                at IndexingStatusResolver.<anonymous> (/opt/indexer/packages/indexer-common/dist/indexing-status.js:210:31)
                at Generator.next (<anonymous>)
                at fulfilled (/opt/indexer/packages/indexer-common/dist/indexing-status.js:5:58)
                at runMicrotasks (<anonymous>)
                at processTicksAndRejections (node:internal/process/task_queues:96:5)
        "attemptNumber": 6,
        "retriesLeft": 0
      },
      "attemptNumber": 6,
      "retriesLeft": 0
    }
[23:13:50.228] WARN (IndexerAgent/1): Failed to obtain protocol chain's latest valid epoch
    component: "Agent"
    error: {
      "code": "IE069",
      "explanation": "https://github.com/graphprotocol/indexer/blob/main/docs/errors.md#ie069",
      "cause": {
        "code": "IE070",
        "explanation": "https://github.com/graphprotocol/indexer/blob/main/docs/errors.md#ie070",
        "cause": {
          "attemptNumber": 6,
          "retriesLeft": 0
        },
        "attemptNumber": 6,
        "retriesLeft": 0
      }
    }
```

This is using the same, default endpoint as on v.0.20.8 which I can switch back to and no longer see these errors https://api.thegraph.com/subgraphs/name/graphprotocol/goerli-epoch-block-oracle

**Comment by madumas** [comment]
Regression of #563 ?

**Comment by fordN** [comment]
The issue is that we now rely on graph-node to resolve block hashes for the protocol chain (in the past we just called a function on the smart contracts.)  This update was required to support L2 protocol chains because of some differences with how smart contract function calls are processed there.  

In current code this error will go away if you configure a Goerli RPC endpoint on your graph-node config and make sure that the indexer agent status endpoint is configured to point to an index-node (not a query node because those don't setup RPC endpoints).  

However, we are working on an update today to no longer require you to have Goerli RPC in your graph-node config.  Should be ready in a couple hours.

**Comment by fordN** [comment]
Resolved in #591
